static/_posts/2024-04-01-vgaterm-intro-001.md

```diff
@@ -0,0 +1,152 @@
+
+The details in how we eneded up on our specific design will be elaberated on in another post, but suffice to say we had to coordinate design details across software, hardware, specific part choices, and part pricing.
+
+### Displaying Pixels
+To dispaly pixels, the `CLK_VIS` signal generated by the VGA timing hardware reads bytes from the FIFO and into a Programmable Logic Unit (PLU) which maps the 8 bits into 9 bits, 3 sets of 3 bits for each of the color channels - red, green, and blue. These 9 bits then get sent to the DAC, described in the following section.
+
+An important consideration for this stage is how long it takes to shuttle a byte from the FIFO into the DAC and finally into the monitor. Since there is processing of the read bytes it's possible that the pixels become late compared to `H_SYNC` and `V_SYNC` and this will show up as a horizontal shift. I.E. the "first" pixel could actually show up on the 2nd or 3rd pixel position on the screen, instead of the actual 1st pixel.
+
+{% include figure.html image="/assets/images/left-collum-timing-bug.jpg" caption="Notice the far left column of the screen in white next to the blue. It appears to wrap around from the far right side of the screen because the first pixel of the image is shifted over 2 or 3 positions" width=550 %}
+
+So a method to handle this kind of pixel shifting makes the picture quality better.
+
+### Digital to Analog Converter (DAC)
+Once the pixel byte has been taken out of the FIFO memory it's on its way to being displayed by the monitor. To be displayed this pixel byte must be converted into three analog voltages between 0 and 0.7 volts corresponding to the Red, Green, and Blue color channels. This is accomplished with a [R2R Resistor Ladder DAC]().
+
+An R2R Resistor Ladder DAC is traditionally connected into an op-amp to maintain the output voltage regardless of load. However in our case the load is just the 75 Ohm impedance of the analog color lines in the monitor, so we really only have the one load. One concern, however, is that the resistor ladder can consume a fair amount of current and the lines representing the color coming from an IC may not be rated to supply the amount of current the DAC would need to maintain the voltage.
+
+In earlier versions of the DAC this is what we did (this is what Ben Eater did in his original "Worlds Worst Video Card" video). We did also try an op-amp but that seemed to create some weird artifacts in the colors in images. Lastly what seemed to work best is just feed the input bit lines into line drivers which _could_ supply the appropriate amount of current and now colors are quite vivid.
+
+### IO Control
+To actually function as a terminal Vgaterm must receive serial signals from a connected host on one of the four IO ports and update the video buffer accordingly. It also must handle keyboard input and forward those typed characters to the connected host.
```

Comment:
Four?

Reply:
USB, pin headers, RJ45 (RS-232), and DSUB-9 Serial (RS-232). Are you saying I should put that here too?

Reply:
yeah, I'd say either qualify or omit the number: also maybe call 'em the "terminal IO" ports or something? Since VGA and keyboard are also both IO ports